components/common-go/cgroups/cgroup.go

```diff
@@ -85,3 +87,45 @@ func ReadFlatKeyedFile(path string) (map[string]uint64, error) {
 
 	return kv, nil
 }
+
+func ReadPSIValue(path string) (PSI, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return PSI{}, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	var psi PSI
+	for scanner.Scan() {
+		line := scanner.Text()
+		if err = scanner.Err(); err != nil {
+			return PSI{}, fmt.Errorf("could not read psi file: %w", err)
+		}
+
+		i := strings.LastIndex(line, "total=")
+		if i == -1 {
+			return PSI{}, fmt.Errorf("could not find total stalled time")
+		}
+
+		total, err := strconv.ParseUint(line[i+6:], 10, 64)
+		if err != nil {
+			return PSI{}, fmt.Errorf("could not parse total stalled time: %w", err)
+		}
+
+		if strings.HasPrefix(line, "some") {
+			psi.Some = total
+		}
+
+		if strings.HasPrefix(line, "full") {
+			psi.Full = total
+		}
+	}
```

Comment:
You need to handle the scanner error
https://pkg.go.dev/bufio#example-Scanner-Lines

Reply:
Nevermind, didn't see it already handled on 102. Thanks!